ktorm-support-mysql/src/main/kotlin/me/liuwj/ktorm/support/mysql/Functions.kt

```diff
@@ -167,6 +170,30 @@ fun <T : Any> ColumnDeclaring<T>.ifNull(right: T?): FunctionExpression<T> {
     return this.ifNull(wrapArgument(right))
 }
 
+/**
+ * MySQL sum & if combination, translated to `sum( if(column condition, ifTrue, ifFalse) )`.
+ */
+fun sumIf(condition: BinaryExpression<*>, ifTrue: Int, ifFalse: Int): AggregateExpression<Int> {
+
+    val ifExpr = FunctionExpression(
+            functionName = "if",
+            arguments = listOf(
+                    condition,
+                    ArgumentExpression(ifTrue, IntSqlType),
+                    ArgumentExpression(ifFalse, IntSqlType)
+            ),
+            sqlType = IntSqlType)
+
+    return AggregateExpression(AggregateType.SUM, ifExpr, false, IntSqlType)
+}
+
+/**
+ * MySQL sum with condition, translated to `sum( column condition )`.
+ */
+fun sum(condition: BinaryExpression<*>): AggregateExpression<Int> {
```

Comment:
We already have a `sum` function, see https://github.com/vincentlauvlwj/Ktorm/blob/master/ktorm-core/src/main/kotlin/me/liuwj/ktorm/dsl/Aggregation.kt#L68

Reply:
@vincentlauvlwj I woule like to pass `boolean` expression for `sum` function.
So, I created the function with parameter `BinaryExpression`
Because in my opinion, `sum(column: ColumnDeclaring<C>)` function can't receive expressions like `Employees.salary greaterEq 100L`

Reply:
In a strong-typed design, I think we should not allow to pass `boolean` expressions to `sum` function. Instead, we can have an explicit cast function like `sum((t.salary greaterEq 100L).toInt())` or use the `IF` function to do the same thing `sum(IF(t.salary greaterEq 100L, then = 1, otherwise = 0))`.

You know, in some weakly typed language like C, we can implicitly cast a boolean value to int, but in Kotlin we can't do that, because it's strong-typed, right?

Reply:
And we already have some cast functions, see https://github.com/vincentlauvlwj/Ktorm/blob/master/ktorm-core/src/main/kotlin/me/liuwj/ktorm/dsl/Operators.kt#L496-L531

Reply:
@vincentlauvlwj I got it.
It's necessary to pass casted object instead of direct expression object.